test/e2e/upgrades/topology_manager.go

```diff
@@ -0,0 +1,317 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package upgrades
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"reflect"
+	"regexp"
+	"strconv"
+	"time"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/uuid"
+
+	"k8s.io/client-go/kubernetes/scheme"
+
+	kubeletconfigv1beta1 "k8s.io/kubelet/config/v1beta1"
+	kubeletconfig "k8s.io/kubernetes/pkg/kubelet/apis/config"
+
+	"k8s.io/kubernetes/pkg/kubelet/cm/cpuset"
+	"k8s.io/kubernetes/pkg/kubelet/cm/topologymanager"
+
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2ekubectl "k8s.io/kubernetes/test/e2e/framework/kubectl"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+
+	imageutils "k8s.io/kubernetes/test/utils/image"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+)
+
+const testCmd string = `
+NCPU=$( getconf _NPROCESSORS_ONLN );
+MAXCPU=$( expr $NCPU - 1 );
+echo ALLOWED=$(grep Cpus_allowed_list /proc/self/status | cut -f2);
+for N in $( seq 0 ${MAXCPU} ); do echo "SIBILING$N=$( cat /sys/devices/system/cpu/cpu${N}/topology/core_siblings_list )"; done;
+sleep 12h
+`
+
+// TopologyManagerUpgradeTest tests that the topology manager provides numa-aligned resources
+// before and after a cluster upgrade.
+type TopologyManagerUpgradeTest struct {
+}
+
+// Name returns the tracking name of the test.
+func (TopologyManagerUpgradeTest) Name() string {
+	return "topology-manager-upgrade [sig-node]"
+}
+
+// Skip returns true when this test can be skipped.
+func (TopologyManagerUpgradeTest) Skip(upgCtx UpgradeContext) bool {
+	return false
+}
+
+// Setup creates a pod requesting aligned resources.
+func (t *TopologyManagerUpgradeTest) Setup(f *framework.Framework) {
+	ginkgo.By("Creating a Pod with aligned resources")
+	node := getTopologyManagerSingleNumaNodePolicyEnabledNode(f)
+	if node == nil {
+		framework.Logf("No suitable node configured with Topology Manager before upgrade")
+		return
+	}
+
+	name := "pod-before-" + string(uuid.NewUUID())
+	err := runAndValidate(f, name, node)
+	framework.ExpectNoError(err)
+}
+
+// Test waits for the upgrade to complete, and then verifies that it is possible to run
+// a pod requesting aligned resources after the upgrade
+func (t *TopologyManagerUpgradeTest) Test(f *framework.Framework, done <-chan struct{}, upgrade UpgradeType) {
+	<-done
+	if !isRelevantUpgrade(upgrade) {
+		return
+	}
+
+	ginkgo.By("Verifying it is still possible to create a Pod with aligned resources")
+	node := getTopologyManagerSingleNumaNodePolicyEnabledNode(f)
+	if node == nil {
+		// XXX: but what if the topology manager WAS enabled before the upgrade?
```

Comment:
What is this `XXX`? `NOTE` or something?

Reply:
was a `NOTE` to self which in retrospect is just misleading. Removed.